src/Symfony/Component/Form/Extension/Validator/Constraints/FormValidator.php

```diff
@@ -24,6 +24,13 @@
  */
 class FormValidator extends ConstraintValidator
 {
+    private $resolvedGroups;
```

Comment:
I'm not comfortable with adding a state to any service, this looks like a leak from afar.
But honestly, I don't know this subsystem so I might miss a lot. If we can remove the state, that'd be better IMHO, if it makes sense.

Reply:
Do you see another way to "remember" something during a recursion? Because this validator needs to validate nested fields recursively for each group of a sequence.

Reply:
I now reset it at the end of each sequence.